src/ops.rs

```diff
@@ -0,0 +1,411 @@
+        assert_eq!(array.shape(), &[2, 3]);
+        assert_eq!(array.dtype(), Dtype::Float32);
+
+        array.eval();
+        let data: &[f32] = array.as_slice().unwrap();
+        assert_eq!(data, &[0.0; 6]);
+    }
+
+    #[test]
+    fn test_identity() {
+        let mut array = Array::identity::<f32>(3, StreamOrDevice::default());
+        assert_eq!(array.shape(), &[3, 3]);
+        assert_eq!(array.dtype(), Dtype::Float32);
+
+        array.eval();
+        let data: &[f32] = array.as_slice().unwrap();
+        assert_eq!(data, &[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]);
+    }
+
+    #[test]
+    fn test_linspace_int() {
+        let mut array = Array::linspace::<f32, _>(0, 50, None, StreamOrDevice::default());
+        assert_eq!(array.shape(), &[50]);
+        assert_eq!(array.dtype(), Dtype::Float32);
+
+        array.eval();
+        let data: &[f32] = array.as_slice().unwrap();
+        let expected: Vec<f32> = (0..50).map(|x| x as f32 * (50.0 / 49.0)).collect();
+        assert_eq!(data, expected.as_slice());
+    }
+
+    #[test]
+    fn test_linspace_float() {
+        let mut array = Array::linspace::<f32, _>(0., 50., None, StreamOrDevice::default());
+        assert_eq!(array.shape(), &[50]);
+        assert_eq!(array.dtype(), Dtype::Float32);
+
+        array.eval();
+        let data: &[f32] = array.as_slice().unwrap();
+        let expected: Vec<f32> = (0..50).map(|x| x as f32 * (50.0 / 49.0)).collect();
+        assert_eq!(data, expected.as_slice());
+    }
+
+    #[test]
+    fn test_repeat() {
+        let source = Array::from_slice(&[0, 1, 2, 3], &[2, 2]);
+        let mut array = Array::repeat::<i32>(source, 4, 1, StreamOrDevice::default());
+        assert_eq!(array.shape(), &[2, 8]);
+        assert_eq!(array.dtype(), Dtype::Int32);
+
+        array.eval();
+        let data: &[i32] = array.as_slice().unwrap();
+        assert_eq!(data, [0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3]);
+    }
+
+    #[test]
+    fn test_repeat_all() {
+        let source = Array::from_slice(&[0, 1, 2, 3], &[2, 2]);
+        let mut array = Array::repeat_all::<i32>(source, 4, StreamOrDevice::default());
+        assert_eq!(array.shape(), &[16]);
+        assert_eq!(array.dtype(), Dtype::Int32);
+
+        array.eval();
+        let data: &[i32] = array.as_slice().unwrap();
+        assert_eq!(data, [0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3]);
+    }
+
+    #[test]
+    fn test_tri() {
+        let mut array = Array::tri::<f32>(3, None, None, StreamOrDevice::default());
+        assert_eq!(array.shape(), &[3, 3]);
+        assert_eq!(array.dtype(), Dtype::Float32);
+
+        array.eval();
+        let data: &[f32] = array.as_slice().unwrap();
+        assert_eq!(data, &[1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 1.0]);
+    }
+
+    #[test]
+    fn test_abs() {
+        let data = [1i32, 2, -3, -4, -5];
+        let array = Array::from_slice(&data, &[5]);
```

Comment:
Maybe we can add a test to see if the original array is still valid after evaluating `result`